lib/fs.js

```diff
@@ -2051,7 +2053,29 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     }
     return;
   }
-  // write(fd, buffer, offset, length, position, callback)
+  if (isCustomIterable(buffer)) {
+    (async () => {
+      for await (const buf of buffer) {
+        fs.write(
+          fd, buf, undefined,
+          isArrayBufferView(buf) ? buf.byteLength : encoding,
+          null, (writeErr, _) => {
+            if (writeErr) {
+              if (isUserFd) {
+                callback(writeErr);
+              } else {
+                fs.close(fd, (err) => {
+                  callback(aggregateTwoErrors(err, writeErr));
+                });
+              }
```

Comment:
I think that in general it would make more sense to continue iterating only after a write has finished.

Reply:
Agreed. As it is this will have issues with backpressure, risking flooding the destination with pending write requests. Also, I'm missing the bit about exiting the `for await` if the `fs.write()` errors on any single iteration.